src/relay/backend/te_compiler.cc

```diff
@@ -224,23 +271,28 @@ class TECompilerImpl : public TECompilerNode {
     }
     cur_ccache_key_ = key;
 
-    // No need to lower external functions for now. We will invoke the external
-    // codegen tool once and lower all functions together.
-    if (key->source_func->GetAttr<String>(attr::kCompiler).defined()) {
-      auto ir_module = IRModule();
-      const auto name_node = key->source_func->GetAttr<String>(tvm::attr::kGlobalSymbol);
-      ICHECK(name_node.defined()) << "External function has not been attached a name yet.";
-      auto func_name = GetUniqueName(name_node.value(), &name_map_);
+    Optional<String> opt_compiler = key->source_func->GetAttr<String>(attr::kCompiler);
+    if (opt_compiler.defined()) {
+      // Don't compile now since we don't have anywhere to put the resulting runtime module.
+      // Instead place the original definition in the cache and wait for LowerExternalFunctions.
+      IRModule ir_module;
+      // Note that the source_func may already be bound to a global function in the module
```

Comment:
In the code you ICHECK that opt_global_symbol is defined, so I'd suggest changing "may" to "should be" or "is"

Reply:
I moved the comment down a bit -- it is really talking about that we are using opt_global_symbol.value() as *the* global name and not attempting to rename.